### DISCUSSION TITLE
Debugging traces not working

### DISCUSSION THREAD

**Discussion by davecates** [BowlerHatLLC/vscode-as3mxml#752] [General]
Hi,

I've got VS code setup correctly but cannot get the trace out from the connected devices - either via wifi or USB.

I have the launch.json and asconfig.json files setup as in the docs for USB debugging. I have also correctly entered the IP address of my development machine when asked by the Air prompt.

It was previously working...any ideas? Any issues with updates to VS Code or Air SDK?

Air: 51.0.1 or any recent version is the same

Thanks.

**Comment by Adolio** [comment]
Did you try connecting via Adobe Scout?

**Comment by davecates** [comment]
Yes, finally got Scout working - took ages to find it and work out how to configure it both on the iOS device and Mac development machine. But alas, no real improvement on the trace outs...any ideas?

**Reply by joshtynjala** [reply]
For some reason, I could not get USB debugging working either. It was working the last time that I tried, and I haven't changed anything in vscode-swf-debug recently. However, maybe iOS has changed how it grants permissions needed for that, and AIR needs to fix something to make it work again.

I can get Wi-fi debugging working, though.

In my asconfig.json, I basically have this for my iOS AIR options for Wi-fi debugging:

```jsonc
"airOptions": {
	"ios": {
		"signingOptions": {
			"storetype": "pkcs12",
			"keystore": "path/to/certificate.p12",
			"provisioning-profile": "path/to/file.mobileprovision"
		},
		"output": "bin/MyApp.ipa"
	}
}
```

(it should also work with `"storetype": "KeychainStore"` and `"alias": "Apple Development: My Company (XXXXX)"` instead, but I've been exporting a .p12 for over a decade, so that's what I've stuck with)

And this is my launch configuration:

```jsonc
{
	"type": "swf",
	"request": "attach",
	"name": "Install & Attach Adobe AIR (iOS)",
	"platform": "ios",
	"internalConsoleOptions": "openOnSessionStart"
}
```

I ran the **Adobe AIR: package iOS debug - asconfig.json** task. Then, after it completed successfully, I started the launch configuration.

When it finishes installing the app on my iOS device, I manually launched the app as explained in the Debug Console. iOS asked for permission to look for local devices on my network. I pressed Allow. Then, a Flash Debugger dialog appears. This dialog did not appear when I successfully debugged on an iOS device from VSCode in the past; the app knew what to connect to automatically. Again, I think iOS may have changed how permission needs to be granted, and AIR is not handling that well.

However, I found two different things that worked for me.

- Manually type my computer's IPv4 address (like 192.168.x.x) and pressing OK.
- Press Cancel. Then, close the app completely, and start app again fresh from the home screen. It automatically connected, like it used to in the past.

EDIT: Followup because I was able to get USB debugging working. It seems that AIR's default port value of 7936 for USB forwarding is not working on my Mac, but specifying a custom port worked for me. Maybe macOS has a new service on 7936 or something that it didn't in the past. Or maybe some new restriction on certain ranges of ports that it didn't have before.

iOS AIR options in asconfig.json for USB debugging (with `listen` option set to a port number instead of just `true`):

```jsonc
"airOptions": {
	"ios": {
		"signingOptions": {
			"storetype": "pkcs12",
			"keystore": "path/to/certificate.p12",
			"provisioning-profile": "path/to/file.mobileprovision"
		},
		"listen": 65531,
		"output": "bin/MyApp.ipa"
	},
}
```

Launch configuration (with `connect` option set to `true` and `port` option set to same port as above):

```jsonc
{
	"type": "swf",
	"request": "attach",
	"name": "Install & Attach Adobe AIR (iOS)",
	"connect": true,
	"port": 65531,
	"platform": "ios",
	"internalConsoleOptions": "openOnSessionStart"
}
```

Again, be sure to use the same port number for both.

Same build task as with wi-fi debugging. Also the same: I had to wait for the .ipa to fully install, and then I manually launched it on my device. However, iOS did not ask for any permissions for USB debugging. It automatically connected to VSCode on my computer, and I saw `trace()` output in VSCode's Debug Console right away. No IP address to enter, and no exiting and restarting necessary for USB debugging.

**Reply by davecates** [reply]
Thanks Josh! I'll try the port suggestion...I have exactly the same config and experience as you - also using p12 certs as have done since Air was in beta on desktop - a long time ago!

Will let you know how I get on...thanks again :)
